### PR TITLE
Adding support for tags

### PR DESCRIPTION
**Pull request by seblemaguer**
Hello,

as I have multiple computers and I like classifying packages, I added some tag support. It was also the first time I was playing with extending use-package so I suspect I did a lot of stuff wrong so I do this pull request to get also some feedbacks.

### CODE REVIEW

File: use-package-tags.el [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  package and the list of tags associated the hostname of the
+  current machine is not empty.")
+
+;; 0. Define some helper
+(defun use-package-tags-validate-tags (tags)
+  "Helper to see if the given TAGS may lead to an
+ installation. Returns t if TAGS is nil or the list of tags
+ associated to the current machine hostname is nil. Else, it
+ returns the intersection between the two lists of tags."
+  (let* ((list-tags-cur-host (alist-get (system-name) use-package-tags-enabled nil nil 'string=)))
```
Comment (a13) [inline use-package-tags.el]: afaik, alist-get appeared only in emacs 25.1, so you have to add the dependency to Package-Requires section

File: use-package-tags.el [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+;;
+
+;;; Code:
+
+(require 'use-package-core)
+(require 'dash)
+
+;; Define some machine name identifiers
+(defvar use-package-tags-enabled '()
```
Comment (a13) [inline use-package-tags.el]: why not `defcustom` ?

**Comment by jwiegley** [comment]
The only reason I hesitate to add this, is that you could already achieve this using `:if`:

```
(defconst this-machine ... some logic to determine a label for this machine ...)

(defun machine-tags (machine) ... a mapping of machines to tags ...)

(defun has-tag (tag) ... returns t if the current machine supports this tag ...)

(use-package foo
  :if (has-tag 'work)
```

I don't know if there need to be another "layer" for something that can be done with Lisp...

**Comment by seblemaguer** [comment]
Thanks a lot for the comment. I propose that I start do that in a package aside to begin and to see what can be the development of this and if it proves useful, I will open a new pull request. Is it ok?

@a13 thanks for the comments I update that on my side

**Comment by jwiegley** [comment]
@seblemaguer If you find that built-in tags support offers better utility or expressiveness, I'll be happy to merge it.

**Comment by seblemaguer** [comment]
I close this pull request for now. I keep you in touch about updates if necessary

**Pull request closed without merging**